src/RequestClientWrapper.ts

```diff
@@ -327,4 +335,33 @@ export default class RequestClientWrapper {
       return stringValue.startsWith(number.toString());
     }
   }
+
+  private isItJwtToken(token: string): boolean {
+    try {
+      JSON.parse(Buffer.from(token.split('.')[1], 'base64').toString());
+    } catch (e) {
+      return false;
+    }
+
+    return true;
+  }
+
+  private async refreshSupabaseSession(): Promise<string> {
+    const supabase = createClient(SUPABASE_URL, SUPABASE_ANON_KEY);
+    const { data, error } = await supabase.auth.setSession({
+      access_token: this.requestClientWrapperConfiguration.session,
+      refresh_token: this.requestClientWrapperConfiguration.session
+    });
+    if (data && data.session) {
```

Comment:
This only refreshes if you need right? And if it doesn't refresh i'm guessing it's empty?

Reply:
@ryankazokas I'm am now checking for expiration and will only refresh when needed.